clients/notification-amqp/src/main/java/org/eclipse/hono/client/notification/amqp/NotificationAddressHelper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.notification.amqp;
+
+import org.eclipse.hono.notification.AbstractNotification;
+import org.eclipse.hono.notification.NotificationConstants;
+import org.eclipse.hono.notification.deviceregistry.AllDevicesOfTenantDeletedNotification;
+import org.eclipse.hono.notification.deviceregistry.CredentialsChangeNotification;
+import org.eclipse.hono.notification.deviceregistry.DeviceChangeNotification;
+import org.eclipse.hono.notification.deviceregistry.TenantChangeNotification;
+
+/**
+ * Utility methods to determine the address for a given type of notification.
+ */
+public final class NotificationAddressHelper {
+
+    private NotificationAddressHelper() {
+        // prevent instantiation
+    }
+
+    /**
+     * Gets the address for sending or receiving notifications.
+     *
+     * @param notificationType The class of the notifications.
+     * @param <T> The type of notifications.
+     * @return The address.
+     * @throws IllegalArgumentException If the given type is not a known subclass of {@link AbstractNotification}.
+     */
+    public static <T extends AbstractNotification> String getAddress(final Class<T> notificationType) {
+        final String address;
+        if (TenantChangeNotification.class.equals(notificationType)) {
+            address = TenantChangeNotification.ADDRESS;
+        } else if (DeviceChangeNotification.class.equals(notificationType)) {
+            address = DeviceChangeNotification.ADDRESS;
+        } else if (CredentialsChangeNotification.class.equals(notificationType)) {
+            address = CredentialsChangeNotification.ADDRESS;
+        } else if (AllDevicesOfTenantDeletedNotification.class.equals(notificationType)) {
+            address = AllDevicesOfTenantDeletedNotification.ADDRESS;
+        } else {
+            throw new IllegalArgumentException("Unknown notification type " + notificationType.getName());
+        }
+
+        return NotificationConstants.NOTIFICATION_ENDPOINT + "/" + address;
```

Comment:
The suffixes (`address` above) are also used in `NotificationTopicHelper`(Kafka case). So, having the suffixes instead of the complete AMQP address or Kafka topic strings defined as constants makes sense IMHO.
Nonetheless, I've extracted the above if block into a common `NotificationConstants#getAddress` method now.

Reply:
But this is the AMQP specific helper class. So can't we define constants like this in it:
```java
private static final String TENANT_CHANGE_ADDRESS = NotificationConstants.NOTIFICATION_ENDPOINT + "/" + TenantChangeNotification.ADDRESS;
...
```
and then return these constant values instead of dynamically creating them?

Moving the code to NotificationConstants doesn't make a difference here ....

Reply:
I've pushed changes.